provider/file/file.go

```diff
@@ -18,68 +20,138 @@ var _ provider.Provider = (*Provider)(nil)
 // Provider holds configurations of the provider.
 type Provider struct {
 	provider.BaseProvider `mapstructure:",squash"`
+	Directory             string `description:"Load configuration from one or more .toml files in a directory"`
 }
 
 // Provide allows the file provider to provide configurations to traefik
 // using the given configuration channel.
 func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
-	watcher, err := fsnotify.NewWatcher()
+	configuration, err := p.loadConfig()
+
 	if err != nil {
-		log.Error("Error creating file watcher", err)
 		return err
 	}
 
-	file, err := os.Open(p.Filename)
+	if p.Watch {
+		var watchDir string
+
+		if p.Directory != "" {
+			watchDir = p.Directory
+		} else {
+			watchDir = filepath.Dir(p.Filename)
```

Comment:
If you just want to watch a file, then you'd just specify the `Filename` and not use `Directory` at all.  So that `Watch` applies only to the active setting `Filename` or `Directory` - they are mutually exclusive.